synapse/app/homeserver.py

```diff
@@ -348,6 +348,22 @@ def setup(config_options: List[str]) -> SynapseHomeServer:
     if config.server.gc_seconds:
         synapse.metrics.MIN_TIME_BETWEEN_GCS = config.server.gc_seconds
 
+    if (
+        config.registration.enable_registration
+        and not config.registration.enable_registration_without_verification
+    ):
+        if (
+            not config.captcha.enable_registration_captcha
+            and not config.registration.registrations_require_3pid
+        ):
```

Comment:
We can probably add `config.registration.registration_requires_token` (token authenticated registration) to this list? 

https://matrix-org.github.io/synapse/latest/usage/administration/admin_api/registration_tokens

Reply:
Sounds fair to me, I'll add it to the list and update the documentation.